Content.Shared/Whitelist/EntityWhitelist.cs

```diff
@@ -1,100 +1,119 @@
+///   - Cigarette
+///   - FirelockElectronics
+///   components:
+///   - Buckle
+///   - AsteroidRock
+///   sizes:
+///   - Tiny
+///   - Large
+/// </code>
+[DataDefinition]
+[Serializable, NetSerializable]
+public sealed partial class EntityWhitelist
 {
     /// <summary>
-    ///     Used to determine whether an entity fits a certain whitelist.
-    ///     Does not whitelist by prototypes, since that is undesirable; you're better off just adding a tag to all
-    ///     entity prototypes that need to be whitelisted, and checking for that.
+    ///     Component names that are allowed in the whitelist.
     /// </summary>
-    /// <code>
-    /// whitelist:
-    ///   tags:
-    ///     - Cigarette
-    ///     - FirelockElectronics
-    ///   components:
-    ///     - Buckle
-    ///     - AsteroidRock
-    /// </code>
-    [DataDefinition]
-    [Serializable, NetSerializable]
-    public sealed partial class EntityWhitelist
-    {
-        /// <summary>
-        ///     Component names that are allowed in the whitelist.
-        /// </summary>
-        [DataField("components")] public string[]? Components = null;
-        // TODO yaml validation
+    [DataField] public string[]? Components;
+    // TODO yaml validation
 
-        [NonSerialized]
-        private List<ComponentRegistration>? _registrations = null;
+    /// <summary>
+    ///     Item sizes that are allowed in the whitelist.
+    /// </summary>
+    [DataField("sizes", customTypeSerializer:typeof(PrototypeIdListSerializer<ItemSizePrototype>))]
+    public List<string>? Sizes;
 
-        /// <summary>
-        ///     Tags that are allowed in the whitelist.
-        /// </summary>
-        [DataField("tags", customTypeSerializer:typeof(PrototypeIdListSerializer<TagPrototype>))]
-        public List<string>? Tags = null;
+    [NonSerialized]
+    private List<ComponentRegistration>? _registrations;
 
-        /// <summary>
-        ///     If false, an entity only requires one of these components or tags to pass the whitelist. If true, an
-        ///     entity requires to have ALL of these components and tags to pass.
-        /// </summary>
-        [DataField("requireAll")]
-        public bool RequireAll = false;
+    /// <summary>
+    ///     Tags that are allowed in the whitelist.
+    /// </summary>
+    [DataField("tags", customTypeSerializer:typeof(PrototypeIdListSerializer<TagPrototype>))]
+    public List<string>? Tags;
 
-        public void UpdateRegistrations()
-        {
-            if (Components == null) return;
+    /// <summary>
+    ///     If false, an entity only requires one of these components or tags to pass the whitelist. If true, an
+    ///     entity requires to have ALL of these components and tags to pass.
+    ///     The "Sizes" criteria will ignores this, since an item can only have one size.
+    /// </summary>
+    [DataField]
+    public bool RequireAll;
+
+    [Dependency] private readonly ILogManager _logMan = default!;
```

Comment:
unused, afaict

Reply:
That was a leftover from my abandoned attempt to swap logging for sawmill. I now went back to the idea because I want the 100% Completion Bonus. Please double check if I used sawmill correctly as I have never touched that before